### PR TITLE
planner, executor, statistics: support correlation calc for new sampling method

### PR DESCRIPTION
**Pull request by time-and-fate**
### What problem does this PR solve?

Issue Number: sub-task of https://github.com/pingcap/tidb/issues/24182

### What is changed and how it works?

add correlation (and extended stats) calculation for the new sampling method

1. Add the `Handle` field in `RowSampleItem`.
2. Calculate `RowSampleItem.Handle` (to support this, we need to add _tidb_rowid to the analyze task if exists).
3. Sort `RowSampleItem` by the handle and assign `Ordinal` to the sample items.
4. Calculate the correlation and extended stats using existed functions.
5. Merge `BuildColumnHistAndTopN` and `BuildHistAndTopNOnRowSample`.
6. Make other codes treat analyze version 3 the same as analyze version 2 (`ver == 2` changed to `ver >= 2`).

Note: There is a small change that correlation of pk column is also calculated, and of course, that's always 1. The correlation of pk column is useless so there isn't any actual influence, but some test result is changed from 0 to 1.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- no release note.

### CODE REVIEW

**Review by winoros — COMMENTED** [review]
lgtm

File: executor/analyze.go [hunk 1]
```diff
@@ -723,24 +733,35 @@ func (e *AnalyzeColumnsExec) buildSamplingStats(ranges []*ranger.Range) (
...
 				sample.Columns[i].SetBytes(sample.Columns[i].GetBytes())
 			}
 		}
+		sample.Handle, err = e.handleCols.BuildHandleByDatums(sample.Columns)
+		if err != nil {
+			return 0, nil, nil, nil, nil, err
+		}
 	}
+
+	sort.Slice(rootRowCollector.Samples, func(i, j int) bool {
```
[the remark this answers is not in the document]
Comment (time-and-fate) [reply executor/analyze.go:752]: Done.

**Review by qw4990 — APPROVED** [review]

**Pull request merged**